### PR TITLE
feat: Support DNS resolution settings

### PR DESCRIPTION
**Pull request by alexwo**
### DNS resolution settings
The purpose is to allow control over DNS resolution settings for non static cluster endpoints. 
With this PR, it will be possible to define custom DNS resolution settings through a BTP policy.

Fixes # https://github.com/envoyproxy/gateway/issues/3690

### CODE REVIEW

File: api/v1alpha1/dns_types.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+// SPDX-License-Identifier: Apache-2.0
+// The full text of the Apache license is available in the LICENSE file at
+// the root of the repo.
+
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+type DNSSettings struct {
+	// DNSRefreshRate specifies the rate at which DNS records should be refreshed.
```
Comment (arkodg) [inline api/v1alpha1/dns_types.go:11]: can we specify the default here ?

File: api/v1alpha1/dns_types.go [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+	DNSRefreshRate *metav1.Duration `json:"dnsRefreshRate,omitempty"`
+	// RespectDNSTTL indicates whether the DNS Time-To-Live (TTL) should be respected.
```
Comment (arkodg) [inline api/v1alpha1/dns_types.go:13]: can we add an extra line in here
```
If the value is set to true, the DNS refresh rate will be set to resource record's TTL
  ```

Comment (arkodg) [reply api/v1alpha1/dns_types.go:13]: also looks like this field defaults to `true` , lets add that info here

File: internal/xds/translator/translator.go [hunk 3]
```diff
@@ -772,6 +772,10 @@ func processXdsCluster(tCtx *types.ResourceVersionTable, httpRoute *ir.HTTPRoute
 		clusterArgs.backendConnection = bt.BackendConnection
 	}
 
+	if httpRoute.DNS != nil {
```
Comment (arkodg) [inline internal/xds/translator/translator.go:775]: is something similar needed for grpc, tcp, udp, tls ?

Comment (alexwo) [reply internal/xds/translator/translator.go:775]: Its probably can be still applicable If the route specifies a DNS hostname, it should function for all protocols within the context of the Envoy gateway cluster.

File: internal/xds/translator/cluster.go [hunk 4]
```diff
@@ -144,6 +145,16 @@ func buildXdsCluster(args *xdsClusterArgs) *clusterv3.Cluster {
 		cluster.ClusterDiscoveryType = &clusterv3.Cluster_Type{Type: clusterv3.Cluster_STRICT_DNS}
 		cluster.DnsRefreshRate = durationpb.New(30 * time.Second)
 		cluster.RespectDnsTtl = true
+		if args.dns != nil {
+			if args.dns.DNSRefreshRate != nil {
+				if args.dns.DNSRefreshRate.Duration > 0 {
+					cluster.DnsRefreshRate = durationpb.New(args.dns.DNSRefreshRate.Duration)
+				}
+				if args.dns.RespectDNSTTL != nil {
```
Comment (guydc) [inline internal/xds/translator/cluster.go:153]: should this live outside the the `args.dns.DNSRefreshRate` nil check, or does envoy require `DNSRefreshRate` to be set for `RespectDNSTTL` to be set?

File: internal/xds/translator/translator.go (hunk 3 above)
Comment (guydc) [reply internal/xds/translator/translator.go:775]: @arkodg  - do you want to create an issue following DNS setting implementation across routes and non-routes? I think that we can start out with limited support for HTTP routes.

File: internal/xds/translator/cluster.go (hunk 4 above)
Comment (alexwo) [reply internal/xds/translator/cluster.go:153]: we default as per above:
	cluster.DnsRefreshRate = durationpb.New(30 * time.Second)
	cluster.RespectDnsTtl = true

File: internal/xds/translator/translator.go (hunk 3 above)
Comment (arkodg) [reply internal/xds/translator/translator.go:775]: yeah for xRoutes is fine for now, what I was trying to get at this, the code is not uniform, for e.g. for TCP Route translation we have another code flow https://github.com/envoyproxy/gateway/blob/cbabdc623acb1c60c9e3790e9e9d870f117eec95/internal/xds/translator/translator.go#L576 so we missed adding it there

Comment (alexwo) [reply internal/xds/translator/translator.go:775]: right, thanks for catching it. I have just added that for tcp and udp routes.

Comment (arkodg) [reply internal/xds/translator/translator.go:777]: thanks @alexwo can you help with the tech debt here , is it possible to reuse `processXdsCluster` for other route types, so we dont miss something like this again

Comment (alexwo) [reply internal/xds/translator/translator.go]: Sure, I have modified processXdsCluster to be compatible with any type that can return cluster arguments,  currently used for TCP, UDP, and HTTP routes.​⬤

**Review by arkodg — APPROVED** [review]
LGTM thanks !

**Review by guydc — APPROVED** [review]

**Pull request merged**